src/onnx/parse_loop.cpp

```diff
@@ -54,7 +54,7 @@ struct parse_loop : op_parser<parse_loop>
             auto arg_iters = args.at(0)->eval();
             if(not arg_iters.empty())
             {
-                max_iterations = arg_iters.at<int64_t>();
+                max_iterations = std::min(arg_iters.at<int64_t>(), max_iterations);
```

Comment:
So the model itself is saying the max iterations should be 2^63?

The purpose of `parser.max_iterations` is to provide a default value when one isn't given in the model, it's not meant to clip the max iterations in the model. We probably need a separate flag for this, like `parser.limit_max_iterations`.

Reply:
Yes, the model has this big number for the loop count.
I will change this and add  `parser.limit_max_iterations` then.